prov/efa/src/rxr/rxr_cq.c

```diff
@@ -726,98 +725,96 @@ bool rxr_cq_need_tx_completion(struct rxr_ep *ep,
 	 * FI_COMPLETION in flags.
 	 */
 	return ep->util_ep.tx_msg_flags == FI_COMPLETION ||
-	       tx_entry->fi_flags & FI_COMPLETION;
+	       op_entry->fi_flags & FI_COMPLETION;
 }
 
 /**
- * @brief write a cq entry for an tx operation (send/read/write) if application wants it.
- *        Sometimes application does not want to receive a cq entry for an tx
- *        operation.
+ * @brief write a cq entry for an operation (send/read/write) if application wants it.
+ *        Sometimes application does not want to receive a cq entry for a tx operation.
  *
  * @param[in]	ep		end point
- * @param[in]	tx_entry	tx entry that contains information of the TX operation
+ * @param[in]	op_entry	tx/rx entry that contains information of the TX operation
  */
 void rxr_cq_write_tx_completion(struct rxr_ep *ep,
-				struct rxr_tx_entry *tx_entry)
```

Comment:
this variable name change is not necessary because we will keep calling this function with tx_entry.

The change of `rxr_tx_entry` to `rxr_op_entry` is good.

It would help if we add an assertion that `assert(tx_entry->type == RXR_TX_ENTRY);`

Reply:
rxr_cq_write_tx_completion() is called inside rxr_cq_handle_tx_completion() which is also called inside rxr_pkt_handle_readrsp_send_completion(). This function sends rx_entry as an argument in this version of the change. Previously it was tx_entry within rxr_pkt_handle_readrsp_send_completion().